### DISCUSSION TITLE
Have to Redeliver Message many times to be successful

### DISCUSSION THREAD

**Discussion by huynhphuong10284** [postalserver/postal#2208] [Help with using Postal]
Hi experts,

I've tried to send email to Gmail and see that some messages delivered successfully and some emails Hard Failed. For failed emails, I use function "Redeliver Message" to re-send emails and see that:

- Some emails just click on Redeliver Message one time and message can be sent successfully
- Some messages have to be clicked many times to send successfully.

I attached a screenshot for your reference.
<img width="577" alt="Screen Shot 2022-10-28 at 09 50 17" src="https://user-images.githubusercontent.com/27182867/198488277-6d472acb-cbd9-4910-8aeb-0b044d6cda96.png">

I'm not sure the issues from Postal or Gmail. For failed email, a error message shows as below:
550-5.7.25 [2a09:2dc0:0:58::55] The IP address sending this message does not 
550-5.7.25 have a PTR record setup, or the corresponding forward DNS entry does 550-5.7.25 not point to the sending IP. As a policy, Gmail does not accept 550-5.7.25 messages from IPs with missing PTR records. Please visit 550-5.7.25 https://support.google.com/mail/answer/81126#ip-practices for more 550 5.7.25 information. bx14-20020a05651c198e00b002772e6ee07dsi1869078ljb.225 - gsmtp

The IP = 2a09:2dc0:0:58::55 some times it shows like above, some time it sends email successfully.
<img width="560" alt="Screen Shot 2022-10-28 at 10 04 27" src="https://user-images.githubusercontent.com/27182867/198489041-7a749e1c-fcf1-4cdf-bf74-72bb00d73c9c.png">

Please review and give me your advices. Thank you very much for your help.

**Comment by willpower232** [comment]
It sounds like you might be missing PTR (reverse DNS) records for one or more of your IPs so it might be worth requesting your server provider or whoever gave you those IP address double check the configuration.

You need them for both IPv4 and IPv6 and Gmail is one of the few that accepts messages on IPv6 so I'd imagine you're missing one there.

**Reply by huynhphuong10284** [reply]
I checked carefully one by one IP which using https://mxtoolbox.com/SuperTool.aspx and all DNS Records found. See screenshot
<img width="1121" alt="Screen Shot 2022-10-28 at 15 25 24" src="https://user-images.githubusercontent.com/27182867/198541432-a55385d1-4dea-46e3-be18-4f4d86dd78ed.png">

**Reply by willpower232** [reply]
fair enough, are these new records? Maybe Google is just having an off day :sweat_smile:

**Reply by huynhphuong10284** [reply]
After take a long time monitoring with IPv6, after you added PTR records - you need to wait 2 months then can send emails via IPv6 (my system with 5k IPv6).

I don't understand why with an email, it's failed and have to retry - it's successful even though the sending on the same IP.
